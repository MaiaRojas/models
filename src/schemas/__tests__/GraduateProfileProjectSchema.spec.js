const mongoose = require('mongoose/browser');
const { GraduateProfileProjectSchema, UserSchema } = require('../')(mongoose);

describe('GraduateProfileProjectSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, GraduateProfileProjectSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate example', (done) => {
    const user = new mongoose.Document({}, UserSchema);
    const graduateProfileProject = new mongoose.Document({
      user: user._id,
      name: 'Foo',
      urls: ['https://lupomontero.github.io/media/'],
    }, GraduateProfileProjectSchema);

    graduateProfileProject.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
