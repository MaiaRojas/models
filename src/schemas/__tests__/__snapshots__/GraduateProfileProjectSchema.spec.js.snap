// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraduateProfileProjectSchema should fail validation when fields missing 1`] = `
Object {
  "name": [ValidatorError: Path \`name\` is required.],
  "urls": [ValidatorError: Validator failed for path \`urls\` with value \`\`],
  "user": [ValidatorError: Path \`user\` is required.],
}
`;
